synapse/cores/common.py

```diff
@@ -604,6 +614,24 @@ def _bumpTufoCache(self, tufo, prop, oldv, newv):
     def _onSetAxonUrl(self, url):
         self.axon = s_telepath.openurl(url)
 
+    def _onSetMods(self, mods):
+        for ctor,conf in mods:
+            # load modules as dyndeps which construct a module
+            # subclass with ctor(<core>,<conf>)
+            try:
+
+                oldm = self.coremods.pop(ctor,None)
+                if oldm != None:
+                    oldm.fini()
+
+                modu = s_dyndeps.tryDynFunc(ctor, self, conf)
```

Comment:
That's kinda a tough problem since the module will wire itself into the events ( and possibly duplicate whatever behavior it's carrying out ) unless it's removed first.  Although, possibly better that than missing an event...   Mostly the use case for these is "on startup" so, i think i'd like to table that issue for now, but we'll def need to think about it if we wanna do dynamic reloads.

Speaking of dynamic behaviors, maybe i should plumb a setModOpt() API that calls a given module's setConfOpt so modules have a supported way to change config options later at runtime.